x/btccheckpoint/README.md

```diff
@@ -0,0 +1,222 @@
+  // - allowing the ZoneConcierge module to prove the checkpoint is submitted to
+  // BTC
+  repeated TransactionInfo txs_info = 2;
+  uint64 epoch = 3;
+}
+```
+
+### Transient States
+
+### BTC Light Client Update
+
+The BTC Light Client Update is maintained in the transient store during block execution. It is accessed using the `BtcLightClientUpdatedKey` and indicates whether the BTC light client was updated during the current block execution.
+
+```go
+func GetBtcLightClientUpdatedKey() []byte {
+	return BtcLightClientUpdatedKey
+}
+```
+
+The module also maintains several key-value pairs in its store:
+
+- `LastFinalizedEpochKey`: Stores the number of the last finalized epoch.
+- `BtcLightClientUpdatedKey`: Indicates whether the BTC light client was updated during the current block execution.
+- `ParamsKey`: Stores module parameters.
+
+These structures and key-value pairs allow the module to efficiently manage and retrieve checkpoint-related data.
+
+## Messages
+
+The BTC Checkpoint module primarily handles messages from vigilantes (also known as submitters or relayers). The message formats are defined in [proto/babylon/btccheckpoint/v1/tx.proto](proto/babylon/btccheckpoint/v1/tx.proto.). The message handlers are likely defined in a file similar to [ x/btccheckpoint/keeper/msg_server.go](x/btccheckpoint/keeper/msg_server.go).  For more information on the SDK messages, refer to the [Cosmos SDK documentation on messages and queries](https://docs.cosmos.network/main/build/building-modules/messages-and-queries)
+
+### MsgInsertBTCSpvProof
+
+`MsgInsertBTCSpvProof` is used by vigilante to insert a new checkpoint into the store, which can be seen [here](https://github.com/babylonlabs-io/vigilante/blob/24da0381465249aa7b55be682a66e32cdaddc81b/types/btccheckpoint.go#L11). 
+
+```protobuf
+message MsgInsertBTCSpvProof {
+  option (cosmos.msg.v1.signer) = "submitter";
+  string submitter = 1;
+  repeated babylon.btccheckpoint.v1.BTCSpvProof proofs = 2;
+}
+```
+
+Upon receiving a `MsgInsertBTCSpvProof`, a Babylon node will execute as follows:
+
+1. Parse and validate the raw checkpoint data from the proof.
+2. Create a new `RawCheckpointSubmission` object from the parsed data.
+3. Verify if the submission is for the expected checkpoint by calling the checkpointing keeper.
+4. Check if the checkpoint is valid for the current epoch.
+5. Verify the ancestors of the checkpoint to ensure continuity.
+6. If all verifications pass, the new checkpoint submission is stored and the checkpoint status is updated.
+
+## MsgUpdateParams
+
+`MsgInsertBTCSpvProof` is used by vigilante to insert a new checkpoint into the store, which can be seen [here](https://github.com/babylonlabs-io/vigilante/blob/24da0381465249aa7b55be682a66e32cdaddc81b/types/btccheckpoint.go#L11).  This message is used to update the `btccheckpoint` module parameters. This should only be executable through governance proposals.
```

Comment:
hmm this is section about `## MsgUpdateParams`, so probably initial sentence is not needed:
```
`MsgInsertBTCSpvProof` is used by vigilante to insert a new checkpoint into the store, which can be seen [here](https://github.com/babylonlabs-io/vigilante/blob/24da0381465249aa7b55be682a66e32cdaddc81b/types/btccheckpoint.go#L11).
```